TempleEight/docs/guide/authentication.md

```diff
@@ -3,3 +3,186 @@ id: authentication
+
+```bash
+# Initial registration
+❯❯❯ curl -X POST $KONG_ENTRY/api/auth/register -d '{"email": "hello@temple.com", "password": "abcdefgh"}'
+{"AccessToken":"..."}
+
+# Subsequent registration
+❯❯❯ curl -X POST $KONG_ENTRY/api/auth/register -d '{"email": "hello@temple.com", "password": "abcdefgh"}'
+{"error":"auth already exists"}
+```
+
+### Login
+The `login` endpoint again requires a POST request with a JSON object containing an `email` and `password` key.
+In response, it will also return an access token that will be used to authenticate subsequent requetss.
+If the email doesn't exist, or the password is incorrect, an error will be returned.
+
+```bash
+# Auth exists and valid credentials provided
+❯❯❯ curl -X POST $KONG_ENTRY/api/auth/login -d '{"email": "hello@temple.com", "password": "abcdefgh"}'
+{"AccessToken":"..."}
+
+# Auth exists, but password is invalid
+❯❯❯ curl -X POST $KONG_ENTRY/api/auth/login -d '{"email": "hello@temple.com", "password": "abcdefghijk"}'
+{"error":"Invalid email or password"}
+
+# Auth doesn't exist
+❯❯❯ curl -X POST $KONG_ENTRY/api/auth/login -d '{"email": "goodbye@temple.com", "password": "abcdefghijk"}'
+{"error":"Invalid email or password"}
+```
+
+## Authenticating Requests to Other Services
+Now that we are able to obtain an access token, we can make requests to other services by including the auth token in the request headers.
+We use the Bearer Token framework, as defined in [RFC6750](https://tools.ietf.org/html/rfc6750).
+
+Let's consider a full example of registration, followed by a request to `ExampleService`:
+
+```bash
+❯❯❯ curl -X POST $KONG_ENTRY/api/auth/register -d '{"email": "test@temple.com", "password": "abcdefgh"}'
+{"AccessToken":"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE1ODgwMTI3OTAsImlkIjoiNGZmYjAyZDQtODdlZC0xMWVhLWE4NjktMDI0MmFjMWYwMDAzIiwiaXNzIjoiaDFNT3hwN2lhTFFOTFA4ek1RS3k2VEkzcmpuNTlsM2MifQ.xNcAdY0r98J7rzNeEWSUPDTUg5HXOCFh41ZB74tTSw0"}
+
+❯❯❯ curl -X POST $KONG_ENTRY/api/example-service -d '{"foo": "hello", "bar": 123}' -H "Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE1ODgwMTI3OTAsImlkIjoiNGZmYjAyZDQtODdlZC0xMWVhLWE4NjktMDI0MmFjMWYwMDAzIiwiaXNzIjoiaDFNT3hwN2lhTFFOTFA4ek1RS3k2VEkzcmpuNTlsM2MifQ.xNcAdY0r98J7rzNeEWSUPDTUg5HXOCFh41ZB74tTSw0"
+{"id":"74045036-87ed-11ea-9edc-0242c0a81003","foo":"hello","bar":123}
+
+```
+Now we included a valid token in the Authorization header, the request was authorized and the entity successfully created.
+
+## Services with `#auth` Metadata
+You may have noticed that so far we have ignored the `ExampleUser` service we previously defined. 
+That's because it acts as any other service, so any example given in this guide could have been replace with a call to that service too.
+
+However, it has one distinguishing feature: a single token is only able to create a single entity in this service. 
+This means means that there is a one-to-one mapping between the `Auth` service, and an entity in the `ExampleUser` service.
```

Comment:
Not sure how to word this better, but it sounds like an ExampleUser is automatically generated for every Auth entry